src/Tempest/Core/src/PublishDiscovery.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Core;
+
+use Tempest\Container\Container;
+use Tempest\Reflection\ClassReflector;
+
+final readonly class PublishDiscovery implements Discovery
```

Comment:
What's still missing is that any class that has this attribute, should be skipped by discovery (this should go in `DiscoveryDiscovery`)

Reply:
That poses a problem though: if we skip all classes with the attribute on a higher level, this discovery class won't ever get any classes to work with…

Reply:
The first point should be fixed by https://github.com/tempestphp/tempest-framework/pull/512. We would need to update relevant classes to use both attributes where necessary.

The second point though... I guess we should add a new kind of discovery that ignores the `HideFromDiscovery` attribute?

Reply:
Shouldn't `#[CanBePublished]` also imply that it's hidden from discovery though? Maybe if we rename it to `#[MustBePublished]`, there would be less confusion?

Yeah the second point… we'll need to hardcode something: `skip discovery if class has #[MustBePublished], UNLESS the current discovery class is `PublishDiscovery`. 

Does that make sense?

Reply:
Another idea: the new `DoNotDiscover` attribute could accept a `allowlist` parameter that would still allow specific discover classes to discover it?

Reply:
I personally think it's fine—I don't mind using multiple attributes, each have their own meaning and there isn't specific logic hidden somewhere else

Reply:
I've given it some more thought, I agree: let's keep them separated!